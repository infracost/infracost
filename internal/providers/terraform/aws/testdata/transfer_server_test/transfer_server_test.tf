provider "aws" {
  region                      = "us-east-1"
  skip_credentials_validation = true
  skip_metadata_api_check     = true
  skip_requesting_account_id  = true
  skip_get_ec2_platforms      = true
  skip_region_validation      = true
  access_key                  = "mock_access_key"
  secret_key                  = "mock_secret_key"
}

resource "aws_transfer_server" "default_no_protocols" {
  tags = {
    Name = "No protocols"
  }
}

resource "aws_transfer_server" "multiple_protocols_with_usage" {
  protocols = ["SFTP", "FTPS", "FTP"]

  tags = {
    Name = "Multiple protocols with usage"
  }
}
