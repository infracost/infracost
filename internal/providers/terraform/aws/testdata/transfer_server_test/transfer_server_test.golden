
 Name                                                 Monthly Qty  Unit            Monthly Cost 
                                                                                                
 aws_transfer_server.default_no_protocols                                                       
 ├─ SFTP protocol enabled                                     730  hours                $219.00 
 ├─ Data downloaded                                 Monthly cost depends on usage: $0.04 per GB 
 └─ Data uploaded                                   Monthly cost depends on usage: $0.04 per GB 
                                                                                                
 aws_transfer_server.multiple_protocols_with_usage                                              
 ├─ FTP protocol enabled                                      730  hours                $219.00 
 ├─ FTPS protocol enabled                                     730  hours                $219.00 
 ├─ SFTP protocol enabled                                     730  hours                $219.00 
 ├─ Data downloaded                                            50  GB                     $2.00 
 └─ Data uploaded                                              10  GB                     $0.40 
                                                                                                
 OVERALL TOTAL                                                                          $878.40 
──────────────────────────────────
2 cloud resources were detected:
∙ 2 were estimated, all of which include usage-based costs, see https://infracost.io/usage-file